usr/sbin/p11sak/p11sak.c

```diff
@@ -1117,6 +1150,151 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
     }
 }
 
+/**
+ *  Print in p11sak_defined_attrs.conf defined attributes in long format
+ */
+static CK_RV print_vendor(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE hkey, int long_print) 
+{
+    CK_RV rc = CKR_OK;
+    int f;
+    struct ConfigBaseNode *c, *name, *hex_string, *type;
+    struct ConfigStructNode *structnode;
+    
+    if (cfg != NULL)
+    {
+        confignode_foreach(c, cfg, f) {
+            if (confignode_hastype(c, CT_STRUCT) && strcmp(c->key, "attribute") == 0) {
+                // parse...
+                structnode = confignode_to_struct(c);
+                name = confignode_find(structnode->value, "name");
+                hex_string = confignode_find(structnode->value, "id");
+                type = confignode_find(structnode->value, "type");
+
+                // checking syntax of attribute...
+                if (name == NULL) {
+                    fprintf(stderr, "Missing name in Attribute in line %hu\n", name->line);
+                    return CKR_DATA_INVALID;
+                } else if (!confignode_hastype(name, CT_BAREVAL)) {
+                    fprintf(stderr, "Invalid name in Attribute in line %hu\n", name->line);
+                    return CKR_DATA_INVALID;
+                } else if (hex_string == NULL) {
+                    fprintf(stderr, "Missing value in Attribute in line %hu\n", hex_string->line);
+                    return CKR_DATA_INVALID;
+                } else if (!confignode_hastype(hex_string, CT_INTVAL)) {
+                    fprintf(stderr, "Invalid name in Attribute in line %hu\n", hex_string->line);
+                    return CKR_DATA_INVALID;
+                } else if (type == NULL) {
+                    fprintf(stderr, "Missing type in Attribute in line %hu\n", type->line);
+                    return CKR_DATA_INVALID;
+                } else if (!confignode_hastype(type, CT_BAREVAL)) {
+                    fprintf(stderr, "Invalid name in Attribute in line %hu\n", type->line);
+                    return CKR_DATA_INVALID;
+                }
+                
+                // print attribute by type
+                unsigned int hex = confignode_to_intval(hex_string)->value;
+                if (strcmp(confignode_to_bareval(type)->value, "CK_BBOOL") == 0) {
+                    // build template
+                    CK_BBOOL a_bool;
+                    CK_ULONG bs = sizeof(CK_BBOOL);
+                    CK_ATTRIBUTE temp = {hex, &a_bool, bs};
```

Comment:
variable `bs` is superfluous, if you simply use
`CK_ATTRIBUTE temp = {hex, &a_bool, sizeof(a_bool)};`

Reply:
Done